src/solvers/smt2/smt2_conv.cpp

```diff
@@ -3448,7 +3505,11 @@ void smt2_convt::convert_with(const with_exprt &expr)
     tmp.operands().resize(s-2);
 
     with_exprt new_with_expr;
-    assert(new_with_expr.operands().size()==3);
+    INVARIANT(
+      new_with_expr.operands().size() == 3,
+      "default constructor of with_exprt creates a version with three "
+      "operands");
```

Comment:
I don't think we need this here, otherwise we'd have to do it in hundreds of places.

What should be done, however, is not use the default constructor, but instead use the one that takes 3 arguments.